src/engine/database/neo4j.rs

```diff
@@ -805,3 +837,56 @@ impl From<Value> for bolt_proto::value::Value {
         }
     }
 }
+
+impl<GlobalCtx, RequestCtx> TryFrom<bolt_proto::value::Value> for Node<GlobalCtx, RequestCtx>
+where
+    GlobalCtx: GlobalContext,
+    RequestCtx: RequestContext,
+{
+    type Error = crate::Error;
+
+    fn try_from(value: bolt_proto::value::Value) -> Result<Self, Error> {
+        match value {
+            bolt_proto::value::Value::Node(n) => {
+                let type_name = &n.labels()[0];
+                let properties: &HashMap<String, bolt_proto::value::Value> = &n.properties();
+                let props_value = Value::try_from(properties.clone())?;
+                let props = match HashMap::<String, Value>::try_from(props_value) {
+                    Ok(v) => v,
+                    Err(_) => {
+                        return Err(Error::TypeConversionFailed {
+                            src: "Value".to_string(),
+                            dst: "HashMap::<String, Value>".to_string(),
+                        })
+                    }
+                };
+                Ok(Node::new(type_name.to_string(), props))
+            }
+            _ => Err(Error::TypeConversionFailed {
+                src: "bolt_proto::value::Value".to_string(),
+                dst: "Node".to_string(),
+            }),
+        }
+    }
+}
+
+impl TryFrom<HashMap<String, bolt_proto::value::Value>> for Value {
+    type Error = Error;
+
+    fn try_from(hm: HashMap<String, bolt_proto::value::Value>) -> Result<Value, Error> {
+        let mut hmv = HashMap::<String, Value>::new();
+        for (key, bolt_value) in hm.iter() {
```

Comment:
I originally had a fold operation on the iterator, but there does not appear to be a clean way to propagate an error up from a closure in a fold (and since we are performing a try_from, we need to). Do you have any ideas to work around that? (I do prefer a functional style, I just couldn't make it work). 

That being said I would lean against adopting just a single style in the project. Performance and usability always come first, but all other things equal, I personally would defer to individual contributor on style.

Reply:
Ah, there is a solution for that.  Iterators have a try_fold method.

That said, I can take the hint on not going overboard with stylistic comments -- I probably need to get better at distinguishing between what's a genuine readability / maintainability benefit and what's just personal preference of little or no substance and consequence.

Reply:
try_fold is what I was looking for! Fixed.